### PR TITLE
llvm: Make --show-llvm and compile to binary work

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

**Comment by Thirumalai-Shaktivel** [comment]
taken `expr2.py` as an input file
```console
✗ python a.py                           
def main():
    x: i32
    x2: i64
    y: f32
    y2: f64
    x = (2 + 3) * 5
    print(x)

Module(body=[FunctionDef(name='main', args=arguments(posonlyargs=[], args=[], kwonlyargs=[], kw_defaults=[], defaults=[]), body=[AnnAssign(target=Name(id='x', ctx=Store()), annotation=Name(id='i32', ctx=Load()), simple=1), AnnAssign(target=Name(id='x2', ctx=Store()), annotation=Name(id='i64', ctx=Load()), simple=1), AnnAssign(target=Name(id='y', ctx=Store()), annotation=Name(id='f32', ctx=Load()), simple=1), AnnAssign(target=Name(id='y2', ctx=Store()), annotation=Name(id='f64', ctx=Load()), simple=1), Assign(targets=[Name(id='x', ctx=Store())], value=BinOp(left=BinOp(left=Constant(value=2), op=Add(), right=Constant(value=3)), op=Mult(), right=Constant(value=5))), Expr(value=Call(func=Name(id='print', ctx=Load()), args=[Name(id='x', ctx=Load())], keywords=[]))], decorator_list=[])], type_ignores=[])

<python_ast.Module object at 0x7f14fb78d1f0>

0 0 1 0 4 main 0 0 0 0 0 0 0 6 7 27 1 x 1 27 3 i32 0 0 1 7 27 2 x2 1 27 3 i64 0 0 1 7 27 1 y 1 27 3 f32 0 0 1 7 27 2 y2 1 27 3 f64 0 0 1 5 1 27 1 x 1 2 2 20 2 0 0 20 3 0 2 20 5 0 0 22 16 27 5 print 0 1 27 1 x 0 0 0 0 0 0
```
**--show-llvm**:
```console
✗ ./src/bin/lfortran --show-llvm ser.txt
; ModuleID = 'LFortran'
source_filename = "LFortran"

@0 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1

define void @main() {
.entry:
  %x = alloca i32, align 4
  %x2 = alloca i64, align 8
  %y = alloca float, align 4
  %y2 = alloca double, align 8
  store i32 25, i32* %x, align 4
  %0 = load i32, i32* %x, align 4
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @0, i32 0, i32 0), i32 %0)
  br label %return

return:                                           ; preds = %.entry
  ret void
}

declare void @_lfortran_printf(i8*, ...)
```
Binaries output:
```console
✗ ./src/bin/lfortran ser.txt; ./a.out   
25
```

**Comment by Thirumalai-Shaktivel** [comment]
This PR is ready to go in!

**Comment by certik** [comment]
Awesome. Would you please mind rebasing on top of the latest main and making sure tests pass?

**Comment by Thirumalai-Shaktivel** [comment]
Sure, I will do that!

**Comment by Thirumalai-Shaktivel** [comment]
Hi @certik, In this PR 2 tests passed. I don't know what the issue is with MACOS test (here is the error: https://github.com/lcompilers/lpython/runs/4856719679?check_suite_focus=true#step:9:4200)
Can you please debug it??

**Comment by certik** [comment]
It seems it created an `expr2` executable and it is trying to run it, but it fails on macOS.

File: ci/build.xsh [hunk 1]
```diff
@@ -83,7 +83,7 @@ cp lpython-$lpython_version/test-bld/src/runtime/*.mod src/runtime/
 # Run some simple compilation tests, works everywhere:
 src/bin/lpython --version
 # Compile and link separately
-src/bin/lpython -c examples/expr2.f90 -o expr2.o
+src/bin/lpython -c examples/expr2.py -o expr2.o
 src/bin/lpython -o expr2 expr2.o
 ./expr2
```
Comment (certik) [inline ci/build.xsh:88]: This line seems to fail on macOS.

**Comment by certik** [comment]
This code is good, let me see if I can figure out how to fix the failure.

File: ci/build.xsh (hunk 1 above)
Comment (certik) [reply ci/build.xsh:88]: The main() function as generated by LLVM did not return 0, so I've fixed that in ASR itself. Hopefully it works now.

**Comment by certik** [comment]
Looks like that was the issue. I am merging this now.

**Comment by Thirumalai-Shaktivel** [comment]
Thank you for debugging and merging this PR!!

**Pull request merged**